### PR TITLE
V3.6.1 oh arm64 support

### PR DESCRIPTION
**Pull request by oahcy**
Re: #

### Changelog

* fix compile flags
* fix windowhandle size

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by oahcy** [comment]
https://github.com/cocos/3d-tasks/issues/13618

**Review by qiuguohua — APPROVED** [review]

**Comment by oahcy** [comment]
depend external： https://github.com/cocos/cocos-engine-external/pull/301

File: native/cocos/platform/interfaces/modules/Device.cpp [hunk 1]
```diff
@@ -47,10 +47,10 @@ uint32_t Device::getInnerHeight() {
     return static_cast<uint32_t>(viewSize.y);
 }
 
-uint32_t Device::getWindowHandle() {
+uintptr_t Device::getWindowHandle() {
     CC_ASSERT(CC_GET_PLATFORM_INTERFACE(ISystemWindow) != nullptr);
     auto handler = CC_GET_PLATFORM_INTERFACE(ISystemWindow)->getWindowHandle();
-    return static_cast<uint32_t>(handler);
+    return static_cast<uintptr_t>(handler);
```
Comment (minggo) [inline native/cocos/platform/interfaces/modules/Device.cpp:53]: Will it work on 32bit if change it like this?

Comment (oahcy) [reply native/cocos/platform/interfaces/modules/Device.cpp:53]: yes

**Review by minggo — APPROVED** [review]

**Pull request merged**